### PR TITLE
Bugg/ti2023 88/Join Button fixes 

### PR DESCRIPTION
**Pull request by bObXm**
Join button is is not displayed for user that are not log in
Join button is disable for user that already join that event so the can only join an event once
Resolved TI2023-89, TI2023-91, TI2023-88

Log in:
Screenshot 2023-11-24 114101

Screenshot 2023-11-24 114413

Log out:
Screenshot 2023-11-24 114213

Screenshot 2023-11-24 114320

### CODE REVIEW

File: openai-assistant-service/env
Comment (munteanumarco) [inline openai-assistant-service/env:1]: Please remove this as it's confidential

**Review by munteanumarco — CHANGES_REQUESTED** [review]

**Review by munteanumarco — APPROVED** [review]
LGTM

File: EventPlannerAPI/EventPlannerAPI/Controllers/UserController.cs [hunk 1]
```diff
@@ -77,6 +77,7 @@ public async Task<IActionResult> ResetPassword([FromBody] ForgotPasswordDto forg
         [ProducesResponseType(StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         [HttpPost("SetNewPassword")]
+
```
Comment (Filip-B-32) [inline EventPlannerAPI/EventPlannerAPI/Controllers/UserController.cs]: empty row between the http and the method

File: event-planner-frontend/src/features/browse-events/thunks/joinEventsThunk.ts [hunk 2]
```diff
@@ -13,7 +13,6 @@ export const joinEventThunk = createAsyncThunk(
       handleGenericSuccess('You have successfully joined the event!');
       return res.data;
     } catch (error: any) {
```
Comment (Filip-B-32) [inline event-planner-frontend/src/features/browse-events/thunks/joinEventsThunk.ts]: i see that you used the toast in the direct component. In my opinion the toast handling for errors should be done just like the success ones

File: event-planner-frontend/src/features/joined-events/JoinedEvents.tsx [hunk 3]
```diff
@@ -16,8 +16,6 @@ const JoinedEvents: React.FC = () => {
   const navigate = useNavigate();
   //   let events: any[] = [];
```
Comment (Filip-B-32) [inline event-planner-frontend/src/features/joined-events/JoinedEvents.tsx]: commented variable

**Review by Filip-B-32 — COMMENTED** [review]
Looks good in the rest

**Review by PhoenixItachi — APPROVED** [review]

**Review by gheorghe-sava — APPROVED** [review]

**Pull request merged**